rawhttp-core/src/test/kotlin/rawhttp/core/RequestLineTest.kt

```diff
@@ -142,19 +142,22 @@ class RequestLineTest {
 
     @Test
     fun canParseEncodedQueries() {
-        val table = table(headers("Request line", "raw query", "decoded query"),
-                row("GET /hi?a=1", "a=1", "a=1"),
-                row("GET /hi?a=1&b=2", "a=1&b=2", "a=1&b=2"),
-                row("GET /hi?a=hi%20w", "a=hi%20w", "a=hi w"),
+        val table = table(headers("Request line", "raw query", "decoded query", "Expected String"),
+                row("GET /hi?a=1", "a=1", "a=1", "GET /hi?a=1 HTTP/1.1"),
+                row("GET /hi?a=1&b=2", "a=1&b=2", "a=1&b=2", "GET /hi?a=1&b=2 HTTP/1.1"),
+                row("GET /hi?a=hi%20w", "a=hi%20w", "a=hi w", "GET /hi?a=hi%20w HTTP/1.1"),
+                row("GET /hi?a=foo%20%26%20bar", "a=foo%20%26%20bar", "a=foo & bar", "GET /hi?a=foo%20%26%20bar HTTP/1.1"),
                 row("GET /hi?%2F%2Fencoded%3Fa%3Db%26c%3Dd&json=%7B%22a%22%3A%20null%7D",
                         "%2F%2Fencoded%3Fa%3Db%26c%3Dd&json=%7B%22a%22%3A%20null%7D",
-                        "//encoded?a=b&c=d&json={\"a\": null}")
+                        "//encoded?a=b&c=d&json={\"a\": null}",
+                        "GET /hi?%2F%2Fencoded%3Fa%3Db%26c%3Dd&json=%7B%22a%22%3A%20null%7D HTTP/1.1")
```

Comment:
I can't believe my test cases failed to pick up this issue!

Reply:
Looks like the test didn't just miss this case, it also missed testing the `toString()` method (even though I do test that in other places - which very unluckily, also failed to check this particular case).